django-backend/fecfiler/f3x_summaries/models.py

```diff
@@ -368,9 +372,72 @@ class F3XSummary(SoftDeleteModel, CommitteeOwnedModel):
     created = models.DateTimeField(auto_now_add=True)
     updated = models.DateTimeField(auto_now=True)
 
+    def save(self, *args, **kwargs):
+        # Record if the is a create or update operation
+        create_action = self.created is None
+
+        with db_transaction.atomic():
+            super(F3XSummary, self).save(*args, **kwargs)
+
+            # Pull forward any loans with non-zero balances along with their
+            # loan guarantors
+            if create_action and self.coverage_through_date:
+                self.pull_forward_loans()
+
+    def pull_forward_loans(self):
+
+        previous_report = (
+            F3XSummary.objects.get_queryset()
+            .filter(
+                committee_account=self.committee_account,
+                coverage_through_date__lt=self.coverage_through_date,
+            )
+            .order_by("coverage_through_date")
+            .last()
+        )
+        if previous_report:
+            loans_to_pull_forward = previous_report.transaction_set.filter(
+                ~Q(loan_balance=Decimal(0.0)) | Q(loan_balance=None),
+                ~Q(memo_code=True),
+                schedule_c_id__isnull=False,
+            )
+
+            for loan in loans_to_pull_forward:
+                # Save children as they are lost from the loan object
+                # when the loan is saved
+                loan_children = copy.deepcopy(loan.children)
+
+                loan.schedule_c_id = self.copy_fkey(loan.schedule_c)
+                loan.memo_text_id = self.copy_fkey(loan.memo_text)
+                loan.report_id = self.id
+                loan.report = self
+                # The loan_id should point to the original loan transaction
+                # even if the loan is pulled forward multiple times.
+                loan.loan_id = loan.loan_id if loan.loan_id else loan.id
+                self.copy_fkey(loan)
```

Comment:
the name `copy_fkey` doesn't really seem to describe what's happening here. what do you think about naming it `copy_record`?

Reply:
yes, that's  better